src/resolvelib/resolvers.py

```diff
@@ -369,11 +377,13 @@ def resolve(self, requirements, max_rounds):
                 # Backtrack if pinning fails. The backtrack process puts us in
                 # an unpinned state, so we can work on it in the next round.
                 success = self._backtrack()
+                self.state.backtrack_causes[:] = [
+                    i for c in failure_causes for i in c.information
+                ]
```

Comment:
Hmm, since `failure_causes` is a list of criteria, maybe `state.backtrack_cause` should be criteria instead? Just nitpicking though, don't need to change it if you don't think it makes sense.

Reply:
I have no opinion, I only wrote it this way as it was the minimal possible change to make this work. So this would be `self.state.backtrack_causes[:] = failure_causes`?

Reply:
On second though, let's not bother now. This is all internal so we can refactor at any time.